gitops/snapshot.go

```diff
@@ -1058,16 +1058,31 @@ func FilterIntegrationTestScenariosWithContext(scenarios *[]v1beta2.IntegrationT
 	return &filteredScenarioList
 }
 
+// UT
+func GetRequiredContextIntegrationTestScenario(scenarios *[]v1beta2.IntegrationTestScenario, contextName string) *[]v1beta2.IntegrationTestScenario {
+	var filteredScenarios []v1beta2.IntegrationTestScenario
+	for _, scenario := range *scenarios {
+		scenario := scenario //G601
+		for _, scenarioContext := range scenario.Spec.Contexts {
+			if scenarioContext.Name == contextName {
+				filteredScenarios = append(filteredScenarios, scenario)
+				continue
+			}
+		}
+	}
+	return &filteredScenarios
+}
+
```

Comment:
For this function I would suggest we use the break approach for better efficiency
also using pointers might cause memory leaks  

```
   func GetRequiredContextIntegrationTestScenario(scenarios []v1beta2.IntegrationTestScenario, contextName string) []v1beta2.IntegrationTestScenario {
	var filteredScenarios []v1beta2.IntegrationTestScenario
	
	for i := range scenarios {
		currentScenario := scenarios[i] // Avoid loop variable capture issue
		
		for _, scenarioContext := range currentScenario.Spec.Contexts {
			if scenarioContext.Name == contextName {
				filteredScenarios = append(filteredScenarios, currentScenario)
				break // Exit inner loop after finding a match
			}
		}
	}
	
	return filteredScenarios
} 
```

Reply:
Thanks! I agree. I wanted to use `break`, but my head doesn't work well at that time...